### PR TITLE
refactor(metrics): suppress more actor-level metrics

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See background in https://github.com/risingwavelabs/risingwave/issues/18108. This PR suppress more actor-level metrics based on metrics level settings.

Some of these metrics already had two panels in actor-level and fragment-level. This PR makes the actor-level one visible by default, but first filtering the `actor_id` to not take suppressed metrics into consideration. So effectively, it's only visible when metrics level is set to `Debug`, typically in development environments.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18318?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#18318** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18318?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#18331** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BugenZhao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/stream/src/executor/monitor/streaming_stats.rs [hunk 1]
```diff
@@ -1044,26 +1049,29 @@ impl StreamingMetrics {
...
         )
-        .unwrap();
+        .unwrap()
+        .relabel_debug_1(level);
 
         let stream_memory_usage = register_guarded_int_gauge_vec_with_registry!(
             "stream_memory_usage",
             "Memory usage for stream executors",
-            &["table_id", "actor_id", "desc"],
+            &["actor_id", "table_id", "desc"],
```
Comment (BugenZhao) [inline src/stream/src/executor/monitor/streaming_stats.rs:1070]: Note the changes on the label ordering. As we always suppress a prefix of this slice, the front one should have the largest cardinality.

File: e2e_test/nexmark/create_sources.slt.part [hunk 2]
```diff
@@ -36,7 +36,7 @@ CREATE SOURCE nexmark (
 ) WITH (
     connector = 'nexmark',
     nexmark.split.num = '2',
-    nexmark.min.event.gap.in.ns = '100'
+    nexmark.min.event.gap.in.ns = '100000000'
```
Comment (kwannoel) [inline e2e_test/nexmark/create_sources.slt.part:39]: Is this change is to avoid too much metrics when debugging the metrics service with a nexmark source?

**Review by kwannoel — APPROVED** [review]
LGTM, thanks.

Do we need to change anything on the cloud side? I suppose not, since the `actor_id` labels are still present, just set to `""`. So it should be backwards compatible with the existing cloud promqls.

File: e2e_test/nexmark/create_sources.slt.part (hunk 2 above)
Comment (BugenZhao) [reply e2e_test/nexmark/create_sources.slt.part:39]: Pushed mistakenly. 🤣 Will revert.

**Comment by BugenZhao** [comment]
> Do we need to change anything on the cloud side? I suppose not, since the `actor_id` labels are still present, just set to `""`. So it should be backwards compatible with the existing cloud promqls.

`avg` needs to be rewritten into `sum(value) / sum(count)`, if any. I will confirm with @arkbriar.

**Comment by BugenZhao** [comment]
> > Do we need to change anything on the cloud side? I suppose not, since the `actor_id` labels are still present, just set to `""`. So it should be backwards compatible with the existing cloud promqls.
> 
> `avg` needs to be rewritten into `sum(value) / sum(count)`, if any. I will confirm with @arkbriar.

Need to update the query for back-pressure rate. However it's currently not version-aware while `stream_actor_count` only exists in the latest version. 🤡

https://github.com/risingwavelabs/risingwave-cloud/blob/d7ba02e1460145b2f6642caae6718f2d9d806630/mgmt/internal/infra/metrics/promquery/types.go#L188-L195

**Pull request merged**